packages/fx-core/src/common/tools.ts

```diff
@@ -945,3 +947,32 @@ export async function getSPFxToken(
   }
   return spoToken;
 }
+
+export function getFixedCommonProjectSettings(rootPath: string | undefined) {
+  if (!rootPath) {
+    return undefined;
+  }
+
+  try {
+    const projectSettingsPath = path.join(
+      rootPath,
+      `.${ConfigFolderName}`,
+      InputConfigsFolderName,
+      ProjectSettingsFileName
+    );
+
+    if (!projectSettingsPath || !fs.pathExistsSync(projectSettingsPath)) {
+      return undefined;
+    }
+
+    const projectSettings = fs.readJsonSync(projectSettingsPath);
+    return {
+      projectId: projectSettings?.projectId ?? undefined,
+      isFromSample: projectSettings?.isFromSample ?? undefined,
+      programmingLanguage: projectSettings?.programmingLanguage ?? undefined,
+      hostType: projectSettings?.solutionSettings?.hostType,
```

Comment:
no need fallback undefined for hostType?

Reply:
Fixed, thx!